spec/requirefire.spec.ts

```diff
@@ -49,6 +49,12 @@ describe("requirefire", () => {
     expect(outer.inner.now).not.toEqual(requiredInner.now);
   });
 
+  test("transitive requires have require extensions and resolve", () => {
+    const outer = _require("./fixtures/outer_transitive");
+    expect(outer.inner.requireKeys).toContain("extensions");
+    expect(outer.inner.requireKeys).toContain("resolve");
+  });
```

Comment:
I think it might be good to test that require.resolve actually works inside these files, assigning it I think is all that is required but there might be some funky stuff it does internally such that we should exercise it.